.circleci/config.yml

```diff
@@ -1,30 +1,29 @@
 version: 2
 jobs:
-  build:
+  build_and_test:
     docker:
-      - image: circleci/python:3.6.1
-    working_directory: ~/repo
+      - image: circleci/python:3.6.4
+      - image: trufflesuite/ganache-cli:v6.4.1
+        command: -d -k=petersburg -i 1001
+    working_directory: ~/build
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - v1-dependencies-{{ checksum "requirements.txt" }}
-            - v1-dependencies-
       - run:
-          name: install dependencies
-          command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip install -r requirements.txt
+          name: Install Dependencies
+          command: sudo pip install -r requirements.txt
+      - run:
+          name: Lint
+          command: flake8
+      - run:
+          name: Test
+          command: tox
       - save_cache:
+          key: cache-{{ .Environment.CIRCLE_SHA1 }}
           paths:
-            - ./venv
-          key: v1-dependencies-{{ checksum "requirements.txt" }}
-      - run:
-          name: run tests
-          command: |
-            . venv/bin/activate
-            python manage.py test
-      - store_artifacts:
-          path: test-reports
-          destination: test-reports
+            - ~/build
+
+workflows:
+  version: 2
+  build_and_test:
+    jobs:
+      - build_and_test
```

Comment:
Why are you doing it all in one task? Too hard to figure out the cacheing and stuff? The advantage of doing it in multiple tasks is (1) you can do everything in parallel (if you have multiple steps like linting, coverage, etc) (2) everything runs even if one step fails (so you could see both the linting and testing failed for example)